### DISCUSSION TITLE
Obtaining environment-wide ICoreWebView2CookieManager without creating a view instance

### DISCUSSION THREAD

**Discussion by nnemkin** [MicrosoftEdge/WebView2Feedback#3875] [Q&A]
As far as I understand, cookie storage is shared between all views in the same environment. I'd like to set and inspect cookies in the environment before creating any views.

Questions:
1. Is there a way to get environment-wide ICoreWebView2CookieManager?
2. If I create a dummy view just to get it's cookie manager, will the view be retained until I release the cookie manager?
3. Do cookie changes from a particular ICoreWebView2CookieManager immediately propagate to all views from the same environment?

**Comment by peiche-ms** [comment]
Hello @nnemkin,

In regards to the questions,

1. With our current API design, you have to obtain an ICoreWebView2CookieManager via [ICoreWebView2_2::get_CookeManager](https://docs.microsoft.com/microsoft-edge/webview2/reference/win32/icorewebview2_2#get_cookiemanager) from any WebView that shares the same environment.
2. WebView should be retained until you close it, and the cookie manager will be released then.
3. Yes this is correct.

**Comment by darbid** [comment]
I may be slightly hijacking this question but in a .NET app I have a similar issue/question with cookies.  Maybe, @peiche-jessica you might let us know the concept I should follow.  My use case is that I target a website that has session cookies. I have learnt #1167 that session cookies are not saved by any browser, which means, for Webview2 I will need to save and add them back if the user for some reason closes my app. 

Currently one would need to wait until a CoreWebView2 instance exists before adding cookies. After the instance exists I can then add the cookies and navigate to the site or refresh if the source property is set.  Is this intended?  I might be being unreasonable, but I feel that the setting up of cookies is better connected to the "environment" of WebView2.

**Comment by rgup63** [comment]
Hi, @peiche-jessica
Is there any way to persist the cookies after the last webview2 closes?

We are explicitly disposing of the webviews on tab/window close events, and it stops all processes if that’s the last view being closed, and because of this, all the cookies are getting destroyed.  Is there any way we can store the cookie data and use it after the last webview closes?
